scripts/device-catalog/index.js

```diff
@@ -89,29 +124,112 @@ function getBoardVerification(board) {
     };
 }
 
-function getBoardMetadata(arch, boardId, suffix) {
+function parseBoardYml(boardYml) {
+    let boards = [];
+    let name = boardYml.name;
+    let vendor = boardYml.vendor || null;
+    let revision = null;
+
+    if ('revision' in boardYml) {
+        revision = boardYml.revision.default;
+    }
+
+    if ('socs' in boardYml) {
+        if (boardYml.socs.length > 1) {
+            for (soc of boardYml.socs) {
+                boards.push({ 'id':`${name}_${soc.name}`, 'soc':soc.name, 'rev':revision, 'vendor':vendor });
+            }
+        } else {
+            boards.push({ 'id':name, 'soc':boardYml.socs[0].name, 'rev':revision, 'vendor':vendor });
+        }
+    } else {
+        boards.push({ 'id':name, 'socs':null, 'rev':revision, 'vendor':vendor });
+    }
+
+    return boards;
+}
+
+function getBoardInfo(boardPath) {
+    const yaml = YAML.parse(fs.readFileSync(path.join(boardsRoot, boardPath,
+        'board.yml'), 'utf8'))
+
+    let boards = [];
+
+    if ("boards" in yaml) {
+        for (const boardYml of yaml.boards) {
+            boards.push(...parseBoardYml(boardYml));
+        }
+        return boards
+    }
+
+    // Single variant boards (majority of boards)
+    boards.push(...parseBoardYml(yaml.board));
+
+    return boards
+}
+
+function getSpecificYaml(board) {
+    let yamlName = `${board.boardId}.yaml`;
+    let yamlPath = path.join(boardsRoot, board.path, yamlName);
+    if (fs.existsSync(yamlPath)) { return yamlPath; }
```

Comment:
JS convention for locally repeated behavior like this is to make a function within the current scope:

```js
const getFileIfExists = (yamlName) => {
    const yamlPath = path.join(boardsRoot, board.path, yamlName);
    if (fs.existsSync(yamlPath) return yamlPath
}

let yamlPath = getFileIfExists(`${board.boardId}.yaml`);
if (yamlPath) return yamlPath;

if (board.rev) {
    yamlPath = getFileIfExists(`${board.boardId}_${board.rev.replace(/\./g, '_')}.yaml`);
    if (yamlPath) return yamlPath;
}
// ...
```

Reply:
This is much cleaner now, thank you!